pkg/proxy/oxyproxy.go

```diff
@@ -71,168 +67,42 @@ func (r *OxyRedirect) updateRules(rules []string) {
 // endpoint info based on ip address, as the ip address would be that of the
 // egress proxy i.e host.
-func (r *OxyRedirect) getInfoFromConsumable(ipstr string, info *accesslog.EndpointInfo, srcIdentity policy.NumericIdentity) {
-	ep := r.source
-	ip := net.ParseIP(ipstr)
-	if ip != nil {
-		if ip.To4() != nil {
-			info.IPv4 = ip.String()
-		} else {
-			info.IPv6 = ip.String()
-		}
-	}
-	secIdentity := ep.ResolveIdentity(srcIdentity)
-
-	if secIdentity != nil {
-		info.Labels = secIdentity.Labels.GetModel()
-		info.LabelsSHA256 = secIdentity.Labels.SHA256Sum()
-		info.Identity = uint64(srcIdentity)
-	}
+func (r *OxyRedirect) getInfoFromConsumable(ipstr string,
+	info *accesslog.EndpointInfo, srcIdentity policy.NumericIdentity) {
+	GetInfoFromConsumable(ipstr, info, srcIdentity, r.source)
 }
 
-func (r *OxyRedirect) getSourceInfo(req *http.Request, srcIdentity policy.NumericIdentity) (accesslog.EndpointInfo, accesslog.IPVersion) {
-	info := accesslog.EndpointInfo{}
-	version := accesslog.VersionIPv4
-	ipstr, port, err := net.SplitHostPort(req.RemoteAddr)
-	if err == nil {
-		p, err := strconv.ParseUint(port, 10, 16)
-		if err == nil {
-			info.Port = uint16(p)
-		}
-
-		ip := net.ParseIP(ipstr)
-		if ip != nil && ip.To4() == nil {
-			version = accesslog.VersionIPV6
-		}
-	}
-
-	// At egress, the local origin endpoint is the source
-	if !r.ingress {
-		r.localEndpointInfo(&info)
-	} else if err == nil {
-		if srcIdentity != 0 {
-			r.getInfoFromConsumable(ipstr, &info, srcIdentity)
-		} else {
-			// source security identity 0 is possible when somebody else other than the BPF datapath attempts to
-			// connect to the proxy.
-			// We should log no source information in that case, in the proxy log.
-			log.Warn("Missing security identity in source endpoint info")
-		}
-
-	}
+func (r *OxyRedirect) localEndpointInfo(info *accesslog.EndpointInfo) {
+	LocalEndpointInfo(info, r.source, r.epID)
+}
 
-	return info, version
+func (r *OxyRedirect) getSourceInfo(req *http.Request,
+	srcIdentity policy.NumericIdentity) (accesslog.EndpointInfo, accesslog.IPVersion) {
+	return GetSourceInfo(req.RemoteAddr, srcIdentity, r.ingress, r)
 }
 
+// getDestinationInfo returns the destination EndpointInfo.
 func (r *OxyRedirect) getDestinationInfo(dstIPPort string) accesslog.EndpointInfo {
-	info := accesslog.EndpointInfo{}
-	ipstr, port, err := net.SplitHostPort(dstIPPort)
-	if err == nil {
-		p, err := strconv.ParseUint(port, 10, 16)
-		if err == nil {
-			info.Port = uint16(p)
-		}
-	}
+	return GetDestinationInfo(dstIPPort, r, r.ingress)
+}
 
-	// At ingress the local origin endpoint is the source
-	if r.ingress {
-		r.localEndpointInfo(&info)
-	} else if err == nil {
-		r.egressDestinationInfo(ipstr, &info)
-	}
+// fillReservedIdentity resolves the labels of the specified identity if known
+// locally and fills in the info member fields.
+func (r *OxyRedirect) fillReservedIdentity(info *accesslog.EndpointInfo, id policy.NumericIdentity) {
```

Comment:
This function can be removed and the `proxy.go` version can be called directly

Reply:
done